src/main/java/me/luligabi/coxinhautilities/common/block/dryingrack/DryingRackBlockEntity.java

```diff
@@ -94,6 +94,11 @@ public boolean canPlayerUse(PlayerEntity player) {
         return true;
     }
 
+    @Override
+    public boolean isValid(int slot, ItemStack stack) {
```

Comment:
If i remember correctly, this didn't work with pipes because unlike vanilla hoppers, they can attempt to input more than one item per transaction, rendering this check unhelpful.

Reply:
```suggestion
    @Override
    public int getMaxCountPerStack() {
        return 1;
    }
```

I believe this might work? It was suggested to me previously

Reply:
Wow, I didn’t realize that this method existed. Changing my code to override that as well. I still have to override `isValid`, since `HopperBlockEntity` doesn’t honor `getMaxCountPerStack()` of the inventory it’s inserting into.